apollo-router/src/graphql/mod.rs

```diff
@@ -165,6 +166,20 @@ impl Error {
     }
 }
 
+/// GraphQL spec require that both "line" and "column" are positive numbers.
+/// However GraphQL Java and GraphQL Kotlin return `{ "line": -1, "column": -1 }`
+/// if they can't determine error location inside query.
+/// This function removes such locations from suplied value.
+fn skip_invalid_locations(mut value: Value) -> Value {
+    if let Some(array) = value.as_array_mut() {
+        array.retain(|location| {
+            location.get("line") != Some(&Value::from(-1))
```

Comment:
this only handles the case where the line or column is "-1", what about any other negative number? What happens if we get a float instead?

Reply:
Good point, since spec compliance implies `> 0` to me (since 0 is not a positive number).
Probably better as: 
```
        array.retain(|location| {
            location
                .get("line")
                .map_or(false, |l| l.as_u64().map_or(false, |v| v > 0))
                && location
                    .get("column")
                    .map_or(false, |l| l.as_u64().map_or(false, |v| v > 0))
        })
```
or something like that.

Reply:
Probably need to fix the tests below since I see `0` is being used in them.

Reply:
> this only handles the case where the line or column is "-1", what about any other negative number? What happens if we get a float instead?

@Geal @garypen I don't think we should handle all the cases and just focus on `-1` since it is what graphql-java and graphql-kotlin return.
Those implementations have hardcoded `-1` in their code bases, so there is a 0% chance that we will get other negative values or floats.

At the moment, it's a workaround we added to support customers until they switch to fixed versions of graphql-java and graphql-kotlin. But if we change it to skip all the `<= 0` values or even float, it means we made a conscious decision to relax validation on `locations`.

In my opinion, we shouldn't relax validation and should adhere to the spec. We could add a workaround only in case customers are stuck with a particular graphql implementation to fix this particular case.

Reply:
> Probably need to fix the tests below since I see 0 is being used in them.

@garypen Thanks 👍 
I will fix the tests.

Reply:
In my code I check that both `line` and `column` are `-1` so I'm certain that I can safely remove it without alerting users.

But for other invalid values situation is a different, we silently remove `locations` without users being notified and have an option to fix his subgraph.

Reply:
I do think there's something to be said about adhering to the spec and only adding exceptions where exceptions are warranted/known, rather than being too relaxed.  The spec could/should/may be more clear about saying positive integers rather than numbers (e.g., floats), but it is also somewhat implied that they are integers since the values themselves (`column` and `line`) are meant to point to character offsets.

While impacted versions of this are never going away and we'll probably have to keep this exception in forever, do we have any interest in opening an issue or PR upstream with `graphql-java`?   It could very much be welcome.